snuba/datasets/discover.py

```diff
@@ -0,0 +1,247 @@
+            ('http_method', Nullable(String())),
+            ('http_referer', Nullable(String())),
+            ('exception_stacks.type', Nullable(String())),
+            ('exception_stacks.value', Nullable(String())),
+            ('exception_stacks.mechanism_type', Nullable(String())),
+            ('exception_stacks.mechanism_handled', Nullable(UInt(8))),
+            ('exception_frames.abs_path', Nullable(String())),
+            ('exception_frames.filename', Nullable(String())),
+            ('exception_frames.package', Nullable(String())),
+            ('exception_frames.module', Nullable(String())),
+            ('exception_frames.function', Nullable(String())),
+            ('exception_frames.in_app', Nullable(UInt(8))),
+            ('exception_frames.colno', Nullable(UInt(32))),
+            ('exception_frames.lineno', Nullable(UInt(32))),
+            ('exception_frames.stack_level', Nullable(UInt(16))),
+            ('modules.name', String()),
+            ('modules.version', String()),
+        ])
+
+    def get_transactions_only_columns(self) -> ColumnSet:
+        return ColumnSet([
+            ('trace_id', Nullable(UUID())),
+            ('span_id', Nullable(UInt(64))),
+            ('transaction_hash', Nullable(UInt(64))),
+            ('transaction_op', Nullable(String())),
+            ('start_ts', Nullable(DateTime())),
+            ('start_ms', Nullable(UInt(16))),
+            ('finish_ts', Nullable(DateTime())),
+            ('finish_ms', Nullable(UInt(16))),
+            ('duration', Nullable(UInt(32))),
+        ])
+
+
+class DiscoverDataset(TimeSeriesDataset):
+    """
+    Experimental dataset for Discover that maps the columns of Events and
+    Transactions into a standard format and sends a query to one of the 2 tables
+    depending on the conditions detected.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(
+            dataset_schemas=DatasetSchemas(
+                read_schema=DiscoverSchema(),
+                write_schema=None,
+            ),
+            time_group_columns={
+                'time': 'timestamp',
+                'bucketed_end': 'finish_ts'
+            },
+            time_parse_columns=['timestamp', 'start_ts', 'finish_ts'],
+        )
+
+    def get_query_processors(self) -> Sequence[QueryProcessor]:
+        return [
+            DiscoverProcessor(),
+        ]
+
+    def get_extensions(self) -> Mapping[str, QueryExtension]:
+        return {
+            'project': ProjectExtension(
+                processor=ProjectWithGroupsProcessor(project_column='project_id')
+            ),
+            'timeseries': TimeSeriesExtension(
+                default_granularity=3600,
+                default_window=timedelta(days=5),
+                timestamp_column='timestamp',
+            ),
+        }
+
+    def column_expr(self, column_name, query: Query, parsing_context: ParsingContext):
+        detected_dataset = detect_dataset(query)
```

Comment:
you need to delegate the tags processing to TagColumnProcessor otherwise they will not be resolved.

Reply:
Also could you please depend on the query object to understand which table we are querying `get_data_source`.
The content of the query may have been mutated by the time you get here. The result of detect dataset may not be valid anymore.

Reply:
Currently TagColumnProcessor gets called via `EventsDataset.column_expr` or `TransactionsDataset.column_expr`. Is there a reason we'd rather have this logic in the Discover dataset?

Reply:
I did not notice you were delegating the call to the events and transactions dataset `column_expr` method, which is something we will have eventually to remove. 
Ok, ignore the comment about tags. Please do not depend on detect_dataset here though. This call happens after all query processors have already run, so the query itself may be different.